runner/tasks.py

```diff
@@ -488,6 +488,18 @@ def abort_job(run):
     run.status = RunStatus.ABORTED
     run.save()
 
+def update_commandline_job_status(run, commandline_tool_job_set):
+    job_status_obj = {}
+    for single_commandline_job in commandline_tool_job_set:
+        status = single_commandline_job.pop('status')
+        root = single_commandline_job.pop('root')
+        if status not in job_status_obj:
+            job_status_obj[status] = []
+        job_status_obj[status].append(single_commandline_job)
+
+    run.job_statuses = job_status_obj
+    run.save()
```

Comment:
It is always better to have one save (INSERT) statement in DB. This way you invoke two updates for each Run object for each status check.